[Frontend] minio client in API server can be configured with environment variables

eterna2 opened the pull request:
### Background
I am deploying multiple kubeflow pipelines in a multi-cloud environment (AWS and Azure), and I am using minio-gateway as a proxy to Azure blob storage. 

I need to have configurable minio endpoint to support multiple kubeflow pipelines deployment. Currently, it is hard-coded to the `kubeflow` namespace. 

### Related PR
#1064 is part of my PR to make minio endpoint configurable from the kfp sdk.

### Related issues
#1319

### Changes
This PR exposes the minio parameters as environment variables which allows me to deploy kubeflow pipelines to other namespaces.

- MINIO_ENDPOINT is composed from MINIO_HOST and MINIO_NAMESPACE so that we can do the following in the deployment yaml.

```yaml
env:
  - name: MINIO_NAMESPACE
     valueFrom:
       fieldRef:
         fieldPath: "metadata.namespace"
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/pipelines/1324)
<!-- Reviewable:end -->

IronPan commented:
/lgtm
/approve

IronPan commented:
awesome! thank you

IronPan commented:
/approve cancel

IronPan commented:
do you mind adding a frontend patch in here 
https://github.com/kubeflow/pipelines/tree/master/manifests/namespaced-install
that set the minio namespace env? 
similar to ml-pipeline-persistenceagent-deployment-patch.yaml

eterna2 commented:
@IronPan 
Hey I have rebased with the latest HEAD with the manifest, and added the patch for pipeline-ui.

I also regenerated the `namespaced-install.yaml` with the updated kustomize config.

rileyjbauer commented:
/ok-to-test

IronPan commented:
/lgtm
/approve
